actionpack/lib/action_dispatch/testing/integration.rb

```diff
@@ -283,6 +283,83 @@ def https?
         @https
       end
 
+      # Performs the actual request.
+      def process(method, path, params: nil, headers: nil, env: nil, xhr: false, as: nil)
```

Comment:
@kaspth should it be `Check #process for more details` or `check #process_with_kwargs for more details`? I don't really see the value either way since the code seems pretty self-documenting to me and I'd elect to just move the documentation to `process` and then remove the comments on `post` `patch` `put` `delete` and `head`, but I defer to your expertise.